src/commonTest/kotlin/UuidTest.kt

```diff
@@ -38,34 +45,80 @@ class UuidTest {
     }
 
     @Test
-    fun provides_higher_and_lower_bits() {
+    fun parsing_throws_when_passed_invalid_length_of_string() {
+        assertFailsWith<IllegalArgumentException>(
+            message = "Uuid string has invalid length: c480d6ab-cb0c-427b-a9a6",
+            block = { Uuid.fromString("c480d6ab-cb0c-427b-a9a6") }
+        )
+    }
+
+    @Test
+    fun parsing_throws_when_passed_invalid_format_of_string() {
+        assertFailsWith<IllegalArgumentException>(
+            message = "Uuid string has invalid format: c480d6abcb0c427ba9a619c5f8a146bd",
+            block = { Uuid.fromString("c480d6abcb0c427ba9a619c5f8a146bd") }
+        )
+    }
+
+    @Test
+    fun parsing_throws_when_passed_invalid_characters_in_string() {
+        assertFailsWith<IllegalArgumentException>(
+            message = "Uuid string has invalid characters ghijklmn-opqr-stuv-wxyz-GHIJKLMNOPQR",
+            block = { Uuid.fromString("ghijklmn-opqr-stuv-wxyz-GHIJKLMNOPQR") }
+        )
+    }
+
+    @Test
+    @Suppress("DEPRECATION")
+    fun deprecated_parse_provides_higher_and_lower_bits() {
         val uuid = Uuid.parse("c480d6ab-cb0c-427b-a9a6-19c5f8a146bd")!!
         assertEquals(-6222257497095190851, uuid.leastSignificantBits)
         assertEquals(-4287190811922382213, uuid.mostSignificantBits)
     }
 
-    @Test fun uuid4_generation() {
+    @Test
+    fun provides_higher_and_lower_bits() {
```

Comment:
I took the liberty of making the position of `@Test` consistent in this class. Can move this to a separate commit if you want.